### PR TITLE
[Research] Speed up evaluation for XTREME-S

### PR DESCRIPTION
**Pull request by anton-l**
# What does this PR do?

This adds a couple of improvements to the evaluation parts of the XTREME-S script:
* fix the bug where filtering by language happened multiple times for parallel workers (redundantly)
* use `preprocess_logits_for_metrics` to transform the logits into pred_ids before concatenating them to avoid OOMs
* add the `--language_group` parameter to train on the FLEURS dataset in batches of languages (west/eastern european languages, south asian languages etc.)

Misc:
* add `--ctc_zero_infinity` to handle the noisy FLEURS transcriptions

### CODE REVIEW

**Comment by anton-l** [comment]
@patrickvonplaten these are ready to merge now I think 

Also cc @sanchit-gandhi: the fixes should make your life much easier if you decide to do a run of multilingual translation :)

**Review by patrickvonplaten — APPROVED** [review]
